File: MVC_js/Course.js
class Course {
    constructor(courseID, title, units, description, prerequisities) {
        this.courseID = courseID;
        this.title = title;
        this.units = units;
        this.description = description;
        this.prerequisities = prerequisities;
    }

    get getCourseID() {
        return this.courseID;
    }

    set setCourseID(courseID) {
        this.courseID = courseID;
    }

    get getTitle() {
        return this.title;
    }

    set setTitle(title) {
        this.title = title;
    }

    get getUnits() {
        return this.units;
    }

    set setUnits(units) {
        this.units = units;
    }

    get getDescription() {
        return this.description;
    }

    set setDescription(description) {
        this.description = description;
    }

    get getPrerequisities() {
        return this.prerequisities;
    }

    set setPrerequisities(prerequisities) {
        this.prerequisities = prerequisities;
    }

};

module.exports = Course;

//var a = new Course("CSE110", "Software Engineering", 4, "Best class in UCSD", ["CSE100"]);
//console.log(a.getDescription);
//a.setDescription = "Gary runs this class!";
//console.log(a.getDescription);

File: Course.cpp
//
//  Course.cpp
//  TrytonsPlan
//
//  Created by Dian Yu on 5/11/18.
//  Copyright © 2018 Dian Yu. All rights reserved.
//

#include <iostream>
#include <string>
#include <list>
using namespace std;

class Course {
    
private:
    string courseID;
    string title;
    int units;
    string description;
    list<string> prerequisites;
    
public:
    void setCourseID(string courseID) {
        this->courseID = courseID;
    }
    
    string getCourseID() const {
        return this->courseID;
    }
    
    void setTitle(string title) {
        this->title = title;
    }
    
    string getTitle() const {
        return this->title;
    }
    
    void setUnits(int units) {
        this->units = units;
    }
    
    int getUnits() const {
        return this->units;
    }
    
    void setDescription(string description) {
        this->description = description;
    }
    
    string getDescription() const {
        return this->description;
    }
    
    void setPrerequisites(list<string> prerequisites) {
        this->prerequisites = prerequisites;
    }
    
    list<string> getPrerequisites() const {
        return this->prerequisites;
    }
};
